File: tst/todo/mouse-rect-click.c
#include "pico.h"

int main (void) {
    pico_init(1);

    /* Layer 120x50 with 3 centered buttons equally spaced */
    Pico_Rel_Rect btn1 = { '%', {0.25, 0.50, 0.15, 0.30}, PICO_ANCHOR_C, NULL };
    Pico_Rel_Rect btn2 = { '%', {0.50, 0.50, 0.15, 0.30}, PICO_ANCHOR_C, NULL };
    Pico_Rel_Rect btn3 = { '%', {0.75, 0.50, 0.15, 0.30}, PICO_ANCHOR_C, NULL };

    pico_layer_empty(NULL, "A", (Pico_Abs_Dim){120, 50}, NULL);
    pico_set_layer("A");
    pico_set_color_clear(PICO_COLOR_NAVY);
    pico_output_clear();
    pico_set_draw_color(PICO_COLOR_WHITE);
    pico_output_draw_rect(&btn1);
    pico_output_draw_rect(&btn2);
    pico_output_draw_rect(&btn3);
    pico_set_layer(NULL);

    /* Draw layer at bottom-right, 35%x35% of screen (distorted) */
    Pico_Rel_Rect r = { '%', {0.99, 0.99, 0.35, 0.35}, PICO_ANCHOR_SE, NULL };
    pico_set_color_clear(PICO_COLOR_BLACK);
    pico_output_clear();
    pico_output_draw_layer("A", &r);

    /* Event loop */
    Pico_Event evt;
    while (pico_input_event_timeout(&evt, PICO_EVENT_MOUSE_BUTTON_DN, -1)) {
        if (evt.type == PICO_EVENT_QUIT) {
            break;
        }

        Pico_Mouse win = pico_get_mouse('w', NULL);
        Pico_Mouse pct = pico_get_mouse('%', &r);
        printf(">>> w %4.0f %4.0f | %% %5.3f %5.3f", win.x, win.y, pct.x, pct.y);

        Pico_Rel_Pos pos = { '%', {pct.x, pct.y}, PICO_ANCHOR_NW, NULL };
        if (pico_vs_pos_rect(&pos, &btn1)) {
            printf(" | click 1");
        } else if (pico_vs_pos_rect(&pos, &btn2)) {
            printf(" | click 2");
        } else if (pico_vs_pos_rect(&pos, &btn3)) {
            printf(" | click 3");
        }
        printf("\n");
    }

    pico_init(0);
    return 0;
}

File: lua/tst/todo/mouse-rect-click.lua
require 'pico'

pico.init(true)

-- Layer 120x50 with 3 centered buttons equally spaced
local btn1 = {'%', x=0.25, y=0.50, w=0.15, h=0.30, anc='C'}
local btn2 = {'%', x=0.50, y=0.50, w=0.15, h=0.30, anc='C'}
local btn3 = {'%', x=0.75, y=0.50, w=0.15, h=0.30, anc='C'}

pico.layer.empty('!', nil, "A", {w=120, h=50})
pico.set.layer("A")
pico.set.show { color='navy' }
pico.output.clear()
pico.set.color.draw('white')
pico.output.draw.rect(btn1)
pico.output.draw.rect(btn2)
pico.output.draw.rect(btn3)
pico.set.layer()

-- Draw layer at bottom-right, 35%x35% of screen (distorted)
local r = {'%', x=0.99, y=0.99, w=0.35, h=0.35, anchor='SE'}
pico.set.show { color='black' }
pico.output.clear()
pico.output.draw.layer("A", r)

-- Event loop
while true do
    local e = pico.input.event()
    if e.tag == 'quit' then
        break
    elseif e.tag == 'mouse.button.dn' then
        local win = pico.get.mouse('w')
        local pct = pico.get.mouse('%', r)
        local msg = string.format(">>> w %4.0f %4.0f | %% %5.3f %5.3f", win.x, win.y, pct.x, pct.y)

        local pos = {'%', x=pct.x, y=pct.y}
        if pico.vs.pos_rect(pos, btn1) then
            msg = msg .. " | click 1"
        elseif pico.vs.pos_rect(pos, btn2) then
            msg = msg .. " | click 2"
        elseif pico.vs.pos_rect(pos, btn3) then
            msg = msg .. " | click 3"
        end
        print(msg)
    end
end

pico.init(false)
